### PR TITLE
[Docs] Updating the docs for mesheryctl exp command #11204

### LINKED ISSUE #11204: [Docs] Updating the docs for mesheryctl exp command
### Current State

- [x] There is an '/references/mesheryctl/exp' related md already written, but it is not added to helpers.go.
- [x] 'https://docs.meshery.io/reference/mesheryctl' lacks content related to exp.

### Desired State
- Add docs regarding the subcommand of 'mesheryctl exp'. 

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 📚 [Instructions for contributing to documentation](https://docs.meshery.io/project/contributing/contributing-docs)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### LINKED ISSUE #11238: [mesheryctl] Formatting duplicate code as a function
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

### Current Behavior

https://github.com/meshery/meshery/pull/11225#discussion_r1648874509
There is code written in Meshery that is used in exactly the same format. (In this case, /mesheryctl/pkg/utils)

### Desired Behavior

Extract the duplicates with a function. 

---

### Contributor Resources

- ⌨️ [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by dusdjhyeon**
**Notes for Reviewers**

This PR fixes #11204 
This PR fixes #11238 
/references/mesheryctl

 
image

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 1ed6286e06b79449435b8d88199f934583c525e1 at: https://6674ccc4edf85bf69ead5ba9--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 2f4cf63ccba029e287c03d35bfa97f19e84f582c at: https://6674d30fff107bf2733e505a--meshery-docs-preview.netlify.app

File: docs/pages/reference/mesheryctl-commands.md [hunk 1]
```diff
@@ -831,23 +831,29 @@ Installation, troubleshooting and debugging of Meshery and its adapters
 <thead>
   <tr>
     <th>Command</th>
-    <th>Subcommand</th>
-    <th>Flag</th>
+    <th>Subcommand 1</th>
+    <th>Subcommand 2</th>
```
Comment (leecalcote) [inline docs/pages/reference/mesheryctl-commands.md:835]: @alphaX86  what do you think?

**Comment by leecalcote** [comment]
@singh1203 will you review?

File: docs/pages/reference/mesheryctl-commands.md (hunk 1 above)
Comment (singh1203) [reply docs/pages/reference/mesheryctl-commands.md:835]: Will break the heading consistency

**Review by singh1203 — COMMENTED** [review]
Also, make sure to have other heading consistent

**Comment by dusdjhyeon** [comment]
> Also, make sure to have other heading consistent

You mean the part about subcommand1 and subcommand2? 
If so, what should I do with those when flag replaces subcommand2...

File: docs/pages/reference/mesheryctl-commands.md (hunk 1 above)
Comment (alphaX86) [reply docs/pages/reference/mesheryctl-commands.md:835]: IMO, I think we can just remove the `exp` from table (we can highlight from the heading before table itself) so we can just put it as Command and Subcommand. // @dusdjhyeon

**Review by alphaX86 — COMMENTED** [review]
Changes look good but I need further opinion on table format //@singh1203

File: mesheryctl/pkg/utils/error.go [hunk 2]
```diff
@@ -337,6 +337,28 @@ func formatError(msg string, cmd cmdType) string {
 		return fmt.Sprintf("%s\nSee %s for usage details\n", msg, relationshipUsageURL)
 	case cmdRelationshipGenerateDocs:
 		return fmt.Sprintf("%s\nSee %s for usage details\n", msg, cmdRelationshipGenerateDocs)
+	case cmdComponent:
```
Comment (aabidsofi19) [inline mesheryctl/pkg/utils/error.go:340]: why dont we extract the formatter in a function , right now formatting for each branch is same

**Review by aabidsofi19 — CHANGES_REQUESTED** [review]

File: docs/pages/reference/mesheryctl-commands.md [hunk 3]
```diff
@@ -831,23 +831,29 @@ Installation, troubleshooting and debugging of Meshery and its adapters
...
+    {% for subcommand_hash in command14.subcommands %}{% assign subcomand1 = subcommand_hash[1] %}
       <tr>
+        <td rowspan={{ subcomand1.subcommand.size | plus:1 }} ><a href="{{ site.baseurl }}/reference/mesheryctl/{{ command14.name }}/{{ subcomand.name }}">{{ subcomand1.name }}</a></td>
         <td></td>
-        <td>{{ flag.name }}</td>
-        <td>{{ flag.description }}</td>
+        <td>{{ subcomand1.description}}</td>
       </tr>
+      {% for subcomand_hash in subcomand1.subcommand %}{% assign subcomand2 = subcomand_hash[1] %}
+        <tr>
```
Comment (aabidsofi19) [inline docs/pages/reference/mesheryctl-commands.md:852]:
```suggestion
        <tr>  
```

**Comment by l5io** [comment]
🚀 Preview for commit f1ef25863c599ab44ffceada79179a0821728c36 at: https://66756d68a79a9674ff798ef8--meshery-docs-preview.netlify.app

File: mesheryctl/pkg/utils/error.go (hunk 2 above)
Comment (leecalcote) [reply mesheryctl/pkg/utils/error.go:340]: @alphaX86 ?

**Comment by l5io** [comment]
🚀 Preview for commit 516ab028f3b6962fafc668956abb54dda43432b2 at: https://6676bff4a58cfaa7520137e1--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 8b70daada78e032ccba57cbe03f5dedf30a3d1b9 at: https://6676c1ca235f9b8dd5a5c3e4--meshery-docs-preview.netlify.app

File: mesheryctl/pkg/utils/error.go (hunk 2 above)
Comment (dusdjhyeon) [reply mesheryctl/pkg/utils/error.go:340]: I think it would be better to open a new issue.

Comment (singh1203) [reply mesheryctl/pkg/utils/error.go:340]: IMO it will be ok to open a new issue for this.

**Review by singh1203 — APPROVED** [review]
LGTM.

File: mesheryctl/pkg/utils/error.go (hunk 2 above)
Comment (singh1203) [reply mesheryctl/pkg/utils/error.go:340]: IMO it will be ok to open a new issue for it.

Comment (leecalcote) [reply mesheryctl/pkg/utils/error.go:340]: Agreed

Comment (dusdjhyeon) [reply mesheryctl/pkg/utils/error.go:340]: I'll open and proceed it

**Comment by l5io** [comment]
🚀 Preview for commit 930e4fe165fca3b234618901b09a0d0a49a324cb at: https://6678f2093bf94c1e7e4ac533--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 31e48e7f18894e4b97b8fa05b333edfbfe5109a5 at: https://6678f555faf0e246b90f779b--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit e885092328ee2a7745c6b3def55e56285bd25ce5 at: https://6678fce2a58cfa565f0137a1--meshery-docs-preview.netlify.app

File: mesheryctl/pkg/utils/error.go [hunk 4]
```diff
@@ -195,6 +195,43 @@ func RelationshipsError(msg string, cmd string) string {
 	}
 }
 
+func ComponentSubError(msg string, cmd string) string {
```
Comment (alphaX86) [inline mesheryctl/pkg/utils/error.go:198]: Where exactly this function is referenced?

File: mesheryctl/pkg/utils/error.go [hunk 5]
```diff
@@ -195,6 +195,43 @@ func RelationshipsError(msg string, cmd string) string {
...
+	case "search":
+		return formatError(msg, cmdComponentSearch)
+	case "viewt":
+		return formatError(msg, cmdComponentList)
+	default:
+		return formatError(msg, cmdComponent)
+	}
+}
+
+func ConnectionSubError(msg string, cmd string) string {
```
Comment (alphaX86) [inline mesheryctl/pkg/utils/error.go:211]: Same way here...

File: mesheryctl/pkg/utils/error.go [hunk 6]
```diff
@@ -195,6 +195,43 @@ func RelationshipsError(msg string, cmd string) string {
...
+	case "delete":
+		return formatError(msg, cmdConnectionDelete)
+	case "list":
+		return formatError(msg, cmdConnectionList)
+	default:
+		return formatError(msg, cmdConnection)
+	}
+}
+
+func ExpRelationshipSubError(msg string, cmd string) string {
```
Comment (alphaX86) [inline mesheryctl/pkg/utils/error.go:222]: Here too...

File: mesheryctl/pkg/utils/error.go [hunk 7]
```diff
@@ -244,104 +281,130 @@ func PatternViewError(msg string) string {
...
+		return formatUsageDetails(msg, expRelationshipListURL)
 	default:
 		return fmt.Sprintf("%s\n", msg)
 	}
 }
 
+func formatUsageDetails(msg string, docURL string) string {
+	return fmt.Sprintf("%s\nSee %s for usage details\n", msg, docURL)
+}
+
```
Comment (alphaX86) [inline mesheryctl/pkg/utils/error.go:407]: This looks fine. Good work

**Review by alphaX86 — CHANGES_REQUESTED** [review]

File: mesheryctl/pkg/utils/error.go (hunk 4 above)
Comment (dusdjhyeon) [reply mesheryctl/pkg/utils/error.go]: This is not needed for exp, I was confusing it with another command. Fixed, thanks.

**Comment by l5io** [comment]
🚀 Preview for commit 553a69c103a0bf136c836550aff6d99db019f88e at: https://667a11ee39fd8c6415fa47f0--meshery-docs-preview.netlify.app

**Comment by vishalvivekm** [comment]
@dusdjhyeon 
Let's discuss this in Meshery Dev meeting on Wednesday. Adding this as an agenda item....
Meeting minutes: https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit?pli=1#
Meeting link: https://meet.layer5.io/meshery

**Review by leecalcote — APPROVED** [review]

**Pull request merged**